### PR TITLE
br: adjust file directory `br/pkg/restore`

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: ref #52877

Problem Summary:
Adjust the file directory structure
### What changed and how does it work?
Adjust the file directory structure `br/pkg/restore`. Just move the `function`s and `struct`s.
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/cmd/br/search.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Copyright 2022 PingCAP, Inc. Licensed under Apache-2.0.
 
-package restore
+package main
```
Comment (YuJuncen) [inline br/cmd/br/search.go:3]: It is somehow strange that the `search` command is directly in the main package, maybe create a new package for it?

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

**Review by 3pointer — APPROVED** [review]

File: br/cmd/br/search.go (hunk 1 above)
Comment (Leavrth) [reply br/cmd/br/search.go]: done

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**